Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -192,266 +89,700 @@ void LoadSettings()
-		uint carrierShip;
-		pub::Player::GetShip(carrier_client, carrierShip);
+	if (!shipAlreadyDocked)
+	{
+		std::lock_guard<std::mutex> saveLock(saveMutex);
+
+		nameToCarrierInfoMap[carrierName].dockedShipList.emplace_back(dockingName);
+		nameToDockedInfoMap[dockingName].lastDockedSolar = Players[dockingID].iLastBaseID;
+		nameToDockedInfoMap[dockingName].carrierName = carrierName;
+		idToCarrierInfoMap[carrierID] = &nameToCarrierInfoMap[carrierName];
+		idToDockedInfoMap[dockingID] = &nameToDockedInfoMap[dockingName];
+		idToCarrierInfoMap[carrierID]->lastCarrierLogin = time(nullptr);
+		mobiledockClients[carrierID].iDockingModulesAvailable--;
+	}
 
-		uint clientShip;
-		pub::Player::GetShip(client, clientShip);
+	// Land the ship on the designated base
+	HkBeamById(dockingID, mobileDockingProxyBase);
+	PrintUserCmdText(carrierID, L"Ship docked");
+}
 
-		// Check to see if the carrier is currently in a base. If so, force the client to dock on that base.
-		if (!carrierShip)
+void HkTimerCheckKick()
+{
+	returncode = DEFAULT_RETURNCODE;
+
+	// DELAYEDDOCK vector iteration
+	for (auto& dockdata = dockingInProgress.begin(); dockdata != dockingInProgress.end();)
+	{
+		auto& dd = dockdata->second;
+		const auto& dockingShipID = Players[dd.dockingID].iShipID;
+		if (dockingShipID == 0)
+		{
+			dockdata = dockingInProgress.erase(dockdata);
+			continue;
+		}
+		Vector V1mov, V1rot;
+		pub::SpaceObj::GetMotion(dockingShipID, V1mov, V1rot);
+		if (V1mov.x > 5 || V1mov.y > 5 || V1mov.z > 5)
 		{
+			PrintUserCmdText(dd.dockingID, L"Docking aborted due to craft movement");
+			PrintUserCmdText(dd.carrierID, L"Docking aborted due to craft movement");
+			dockdata = dockingInProgress.erase(dockdata);
+			continue;
+		}
 
-			uint iBaseID;
-			pub::Player::GetBase(carrier_client, iBaseID);
+		Vector V2mov, V2rot;
+		const auto& carrierShipID = Players[dd.carrierID].iShipID;
+		pub::SpaceObj::GetMotion(carrierShipID, V2mov, V2rot);
+		if (V2mov.x > 5 || V2mov.y > 5 || V2mov.z > 5)
+		{
+			PrintUserCmdText(dd.dockingID, L"Docking aborted due to carrier movement");
+			PrintUserCmdText(dd.carrierID, L"Docking aborted due to carrier movement");
+			dockdata = dockingInProgress.erase(dockdata);
+			continue;
+		}
 
-			if (iBaseID)
+		dd.timeLeft--;
+		if (!dd.timeLeft)
+		{
+			Vector pos;
+			Matrix _;
+			pub::SpaceObj::GetLocation(dockingShipID, pos, _);
+			float distance = HkDistance3D(pos, dd.startPosition);
+			if (distance > maxDockingDistanceTolerance)
 			{
-				// Set the flags which the PlayerLaunch_AFTER uses to handle teleporting to the carriers base
-				mobiledockClients[client].undockBase = Universe::get_base(iBaseID);
-				mobiledockClients[carrier_client].mapDockedShips.erase(clientName);
-				mobiledockClients[client].baseUndock = true;
-				return;
+				PrintUserCmdText(dd.dockingID, L"Docking aborted due to craft movement");
+				PrintUserCmdText(dd.carrierID, L"Docking aborted due to craft movement");

```

Comment:
name the craft

Reply:
Given how the whole operation is between two people, you'd know if you moved (because you pressed something) or if the other one moved (because you didn't)

Reply:
see above